File: Ejercicios PLAYWRIGHT/PLAYWRIGHTSAUCEDEMO/src/pages/SauceDemoLoginPage.ts
import {expect, Locator, Page} from '@playwright/test'

export class SauceDemoLoginPage{

    readonly page:Page;
    readonly txtUserName: Locator;
    readonly txtPassword: Locator;
    readonly btnLogin: Locator;
    readonly textSuccess: Locator;
    

    constructor(page:Page)
    {
        this.page = page;
        this.txtUserName = page.locator('[data-test="username"]');
        this.txtPassword = page.locator('[data-test="password"]');
        this.btnLogin =  page.locator('[data-test="login-button"]');
        this.textSuccess = page.getByText('Products');
    }
    
    

    async login(userInfo)
    {
        await this.txtUserName.click();
        await this.txtUserName.fill(userInfo.user);
        await this.txtPassword.click();
        await this.txtPassword.fill(userInfo.password);
        await this.page.waitForTimeout(2000);
        await this.btnLogin.click();
        
    }

    async check()
    {
       await this.textSuccess.isVisible();
    }


    
}

File: Ejercicios POM/POMSAUCEDEMO/pageobjectmodel/src/main/java/pages/SauceDemoLoginPage.java
package pages;


import utils.Constants;
import net.thucydides.core.pages.PageObject;

import java.util.Map;
import org.openqa.selenium.By;
import io.cucumber.datatable.DataTable;

import actions.Action;



public class SauceDemoLoginPage extends PageObject{

    By txtUserName = By.name("user-name");
    By txtPassword = By.name("password");
    By btnLogin = By.name("login-button");
    By textSuccess = By.xpath("//*[text()='Products']");

    public void navegateToUrl(String url){
        
        Action.navegateToUrl(getDriver(),  Constants.getData(url));
    }

    public void login(DataTable dataTable)
    {
        Map<String, String> userInfo = dataTable.asMap(String.class, String.class);
        Action.sendData(getDriver(), txtUserName, userInfo.get("user"));
        Action.sendData(getDriver(), txtPassword, userInfo.get("password"));
        Action.clickElement(getDriver(), btnLogin);

    }
    public void check()
    {
        Action.clickElement(getDriver(), textSuccess);

    }
    
}
